src/qdata/Converters/ExpressionTrees/ExpressionTreesExpressionVisitor.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using RoyLab.QData.Filter.Expressions;
+using RoyLab.QData.Interfaces;
+using RoyLab.QData.Updater.Expressions;
+
+namespace RoyLab.QData.Converters.ExpressionTrees
+{
+    public class ExpressionTreesExpressionVisitor : IExpressionVisitor<Expression>
+    {
+        private readonly ParameterExpression[] parameters;
+
+        public ExpressionTreesExpressionVisitor(params ParameterExpression[] parameters)
+        {
+            this.parameters = parameters;
+        }
+
+        public Expression VisitAndConvert(IExpression expression)
+        {
+            return expression switch
+            {
+                AndExpression andExpression => VisitAndConvert(andExpression),
+                CompareExpression compareExpression => VisitAndConvert(compareExpression),
+                InExpression inExpression => VisitAndConvert(inExpression),
+                NotExpression notExpression => VisitAndConvert(notExpression),
+                OrExpression orExpression => VisitAndConvert(orExpression),
+                AssignExpression assignExpression => VisitAndConvert(assignExpression),
+                _ => null
+            };
+        }
+
+        private Expression VisitAndConvert(AndExpression andExpression)
+        {
+            var leftExpression = andExpression.Left.Accept(this);
+            var rightExpression = andExpression.Right.Accept(this);
+            if (leftExpression == null || rightExpression == null)
+            {
+                return null;
+            }
+
+            return Expression.And(leftExpression, rightExpression);
+        }
+
+        private Expression VisitAndConvert(CompareExpression compareExpression)
+        {
+            var source = parameters.First();
+            var (memberType, memberExpression) = source.AccessPropertyOrMember(compareExpression.Variable);
+            if (memberType == null || memberExpression == null)
+            {
+                return null;
+            }
+
+            Expression valueExpression = Expression.Constant(compareExpression.Value);
+            if (memberType.IsEnum)
+            {
+                var underlyingExpression = TypeUtility.ParseString(valueExpression, Enum.GetUnderlyingType(memberType));
```

Comment:
Does the first parameter of ParseString have to be an Linq expression?  It seems an unnecessary wrapper, why not pass the underlying value